contrib/format-xml/src/main/java/org/apache/drill/exec/store/xml/XMLReader.java

```diff
@@ -298,15 +305,27 @@ private void processEvent(XMLEvent currentEvent,
         } else if (currentState == xmlState.FIELD_ENDED && currentNestingLevel >= dataLevel) {
           // Case to end nested maps
           // Pop tupleWriter off stack
-          currentTupleWriter = rowWriterStack.pop();
-          attributePrefix = XMLUtils.removeField(attributePrefix);
+          if (rowWriterStack.size() > 0) {
+            currentTupleWriter = rowWriterStack.pop();
+          }
+          // Pop field name
+          if (fieldNameStack.size() > 0) {
+            fieldNameStack.pop();
+          }
+
+          attributePrefix = XMLUtils.removeField(attributePrefix,fieldName);
```

Comment:
Could you please confirm that the `attributePrefix` variable will not be null in the context of the `XMLReader.processEvent()` ?

Reply:
> Could you please confirm that the `attributePrefix` variable will not be null in the context of the `XMLReader.processEvent()` ?

@luocooong.  Thanks for the review.

The field `attributePrefix` is defined here:

https://github.com/apache/drill/blob/4e40104d5b4f6ef870f8569edc4981a9eb3ac7ca/contrib/format-xml/src/main/java/org/apache/drill/exec/store/xml/XMLReader.java#L247-L249

If the field `currentNestingLevel` is greater than the `dataLevel` then it will be defined.  If not, then this entire code block will be skipped.